platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java

```diff
@@ -629,65 +630,84 @@ private final BundleStateSet getCurrentBundleStates(Set<Feature> features) {
-     * parameter
-     */
-    private Set<Feature> getNotInstalledFeatures(Set<Feature> features) {
-        Set<Feature> notInstalledFeatures = new HashSet<Feature>();
-        for (Feature curFeature : features) {
-            if (!featuresService.isInstalled(curFeature)) {
-                LOGGER.debug("{} is not installed.", curFeature.getName());
-                notInstalledFeatures.add(curFeature);
-            }
-        }
-        return notInstalledFeatures;
-    }
-
     @Override
     public synchronized void startApplication(Application application)
             throws ApplicationServiceException {
+        final String applicationName = application.getName();
+
         try {
-            LOGGER.debug("Starting Application {} - {}", application.getName(),
-                    application.getVersion());
-            Set<Feature> autoInstallFeatures = application.getAutoInstallFeatures();
-            if (!autoInstallFeatures.isEmpty()) {
-                Set<String> autoFeatureNames = autoInstallFeatures.stream()
-                        .map(Feature::getName)
-                        .collect(Collectors.toSet());
-                for (Feature feature : autoInstallFeatures) {
-                    if (featuresService.isInstalled(feature)) {
-                        autoFeatureNames.remove(feature.getName());
-                    } else {
-                        for (Dependency dependency : feature.getDependencies()) {
-                            if (!application.getName()
-                                    .equals(dependency.getName()) && getApplicationNames().contains(
-                                    dependency.getName())) {
-                                if (!isApplicationStarted(getApplication(dependency.getName()))) {
-                                    startApplication(dependency.getName());
-                                }
-                                autoFeatureNames.remove(dependency.getName());
-                            }
-                        }
-                    }
-                }
-                if (!autoFeatureNames.isEmpty()) {
-                    featuresService.installFeatures(autoFeatureNames,
-                            EnumSet.of(Option.NoAutoRefreshBundles));
-                    waitForApplication(application);
-                }
+            LOGGER.debug("Starting Application {} - {}", applicationName, application.getVersion());
+
+            final Feature mainFeature = application.getMainFeature();
+            if (mainFeature != null) {
+                LOGGER.trace(
+                        "Starting the main feature {} before the rest of the auto-install features in application {}",
+                        mainFeature.getName(),
+                        applicationName);
+                startFeaturesAndDependentAppsAndWait(Collections.singleton(mainFeature),
+                        applicationName);
+            } else {
+                LOGGER.debug(
```

Comment:
I don't think so. There are a few apps included by default that don't have main features.